android/java/org/chromium/chrome/browser/preferences/BravePreferenceFragment.java

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.preferences;
+
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager.NameNotFoundException;
+import android.preference.Preference;
+import android.preference.PreferenceFragment;
+import java.util.HashMap;
+import org.chromium.base.Log;
+import org.chromium.chrome.browser.ChromeVersionInfo;
+import org.chromium.chrome.browser.preferences.PrefServiceBridge.AboutVersionStrings;
+import org.chromium.chrome.browser.preferences.website.SiteSettingsCategory;
+import org.chromium.chrome.browser.preferences.website.SiteSettingsPreferences;
+
+/**
+ * Base class for preferences.
+ */
+public class BravePreferenceFragment extends PreferenceFragment {
+    private static final String PREF_SAFE_BROWSING = "safe_browsing";
+    private static final String PREF_DO_NOT_TRACK = "do_not_track";
+    private static final String PREF_USAGE_AND_CRASH_REPORTING = "usage_and_crash_reports";
+    private static final String PREF_APPLICATION_VERSION = "application_version";
+
+    private static final String VERSION_MASK = "Brave %s, Chromium %s";
+
+    private final HashMap<String, Preference> mRemovedPreferences = new HashMap<>();
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        // Remove preferences that are not used in Brave
+        removePreferenceIfPresent(MainPreferences.PREF_SIGN_IN);
+        removePreferenceIfPresent(MainPreferences.PREF_AUTOFILL_ASSISTANT);
+        removePreferenceIfPresent(MainPreferences.PREF_DATA_REDUCTION);
+        removePreferenceIfPresent(SiteSettingsCategory.preferenceKey(SiteSettingsCategory.Type.ADS));
+        removePreferenceIfPresent(SiteSettingsCategory.preferenceKey(SiteSettingsCategory.Type.BACKGROUND_SYNC));
+        removePreferenceIfPresent(PREF_SAFE_BROWSING);
+        removePreferenceIfPresent(PREF_DO_NOT_TRACK);
+        removePreferenceIfPresent(PREF_USAGE_AND_CRASH_REPORTING);
+
+        // Set proper Brave's version
+        if (this instanceof AboutChromePreferences) {
```

Comment:
it seems like we should be fixing this in the c++ code because we already have overrides for desktop. If `android_build_info->host_package_label()` is returning the wrong value then it seems like we need to fix the source of the issue. Maybe ContextUtils?

Reply:
This will be fixed in separate PR.